### PR TITLE
Fixes for gcc11.2

### PR DESCRIPTION
**Pull request by mike-sul**
Tiny fixes and workarounds to the issues unveiled when aklite and its tests were built and ran in gcc11.2 and clang++ 13.0.1 env.

### CODE REVIEW

**Comment by mike-sul** [comment]
[ci-retest]

File: tests/fixtures/dockerdaemon.cc [hunk 1]
```diff
@@ -9,7 +9,7 @@ class DockerDaemon {
   // tests/docker-compose_fake.py fills/populates this file with "running" containers
   static constexpr const char* const ContainersFile{"containers.json"};
  public:
-  DockerDaemon(boost::filesystem::path dir): dir_{std::move(dir)}, port_{TestUtils::getFreePort()}, process_{RunCmd, "--port", port_, "--dir", dir.string()} {
+  DockerDaemon(boost::filesystem::path dir): dir_{std::move(dir)}, port_{TestUtils::getFreePort()}, process_{RunCmd, "--port", port_, "--dir", dir_.string()} {
```
Comment (doanac) [inline tests/fixtures/dockerdaemon.cc:12]: you just gotta love c++

**Review by doanac — APPROVED** [review]

**Comment by mike-sul** [comment]
[ci-retest]

**Comment by mike-sul** [comment]
[ci-retest]

**Comment by mike-sul** [comment]
@doanac That's all what I wanted to do at this iteration in terms of "dev flow" improvement (see the last commit).
I'll make improvements of aklite running in container in the next iteration.

**Comment by mike-sul** [comment]
[ci-retest]

**Review by doanac — APPROVED** [review]

**Pull request merged**